src/libmongoc/tests/unified/util.c

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2020-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "util.h"
+
+#include "bson-parser.h"
+#include "test-conveniences.h"
+
+mongoc_write_concern_t * bson_to_write_concern (bson_t *bson, bson_error_t* error) {
+    bson_parser_t *parser = NULL;
+    mongoc_write_concern_t *out = NULL;
+    bool *journal;
+    int64_t *w_int;
+    char* w_string;
+    int64_t *wtimeoutms;
+
+    parser = bson_parser_new ();
```

Comment:
I feel like this `bson_parser` is already proving to be so handy!

Reply:
Yeah, I hope it is useful!